test/mnemonic-test.js

```diff
@@ -64,4 +67,127 @@ describe('Mnemonic', function() {
+    const m2 = Mnemonic.fromEntropy(m1.getEntropy(), LANGUAGE_ENGLISH);
+    assert.strictEqual(m1.getPhrase(), m2.getPhrase());
+    assert.strictEqual(LANGUAGE_ENGLISH, m2.language);
+    assert.strictEqual(m1.bits, m2.bits);
+    assert.bufferEqual(m1.getEntropy(), m2.getEntropy());
+  });
+
+  it('should expect an error from fromPhrase() when phrase contains a word not in the wordlist', () => {
+    const m1 = new Mnemonic();
+    const phrase = m1.getPhrase();
+    const phraseArray = phrase.split(' ');
+    phraseArray[1] = 'notaword';
+    const phraseWithBadWord = phraseArray.join(' ');
+
+    assert.throws(() => {
+        Mnemonic.fromPhrase(phraseWithBadWord);
+      },
+      Error,
+      'Unknown word: notaword');
+  });
+
+  it('should expect an error from fromPhrase() when phrase array is missing a word (for some reason)', () => {
+    const m1 = new Mnemonic();
+    const phrase = m1.getPhrase();
+    const phraseArray = phrase.split(' ');
+    phraseArray.pop();
+    const phraseWithMissingWord = phraseArray.join(' ');
+
+    assert.throws(() => {
+        Mnemonic.fromPhrase(phraseWithMissingWord);
+      },
+      Error,
+      'Invalid checksum');
+  });
+
+  it('should handle destroy correctly', () => {
+    const m1 = new Mnemonic();
+    m1.destroy();
+    assert.strictEqual(m1.phrase, null);
+    assert.strictEqual(m1.language, LANGUAGE_ENGLISH);
+    assert.strictEqual(m1.bits, MIN_ENTROPY);
+    assert.strictEqual(m1.entropy, null);
+  });
+
+  it('should handle destroy correctly when entropy is set', () => {
+    const m1 = new Mnemonic();
+    m1.entropy = Buffer.from('00000000000000000000000000000000', 'hex');
+    m1.destroy();
+    assert.strictEqual(m1.phrase, null);
+    assert.strictEqual(m1.language, LANGUAGE_ENGLISH);
+    assert.strictEqual(m1.bits, MIN_ENTROPY);
+    assert.strictEqual(m1.entropy, null);
+  });
+
+  it('should handle fromOptions correctly when a phrase is passed in', () => {
+    const m1 = new Mnemonic();
+    const m2 = Mnemonic.fromOptions(m1.getPhrase().toString());
+
+    assert.strictEqual(m1.phrase, m2.phrase);
+    assert.strictEqual(LANGUAGE_ENGLISH, m2.language);
+    assert.strictEqual(256, m2.bits);
+    assert.strictEqual(32, m2.entropy.length);
+  });
+
+  it('should handle fromOption correctly when options.entropy is set', () => {
+    const entropy = Buffer.from('00000000000000000000000000000000', 'hex');
+    const m2 = Mnemonic.fromOptions({ entropy: entropy});
+
+    assert.strictEqual(null, m2.phrase);
+    assert.strictEqual(LANGUAGE_ENGLISH, m2.language);
+    assert.strictEqual(128, m2.bits);
+    assert.strictEqual(16, m2.entropy.length);
+  });
+
+  it('should handle fromOption correctly when no options are set', () => {
+    const m2 = Mnemonic.fromOptions({ });
+
+    assert.strictEqual(null, m2.phrase);
+    assert.strictEqual(LANGUAGE_ENGLISH, m2.language);
+    assert.strictEqual(256, m2.bits);
+    assert.strictEqual(null, m2.entropy);
+  });
```

Comment:
Take a look at the very tests in this file, I think we are already covered here. And notice in those tests, we pull hard-coded phrases from Trezor's test vectors in two languages. You can add more tests in that section if you think something is missing. I guess you are being explicit about `fromOptions()` being different from using the `new` constructor. Even if that is your goal, I think you can add those tests in that first block

Reply:
I'm shooting for 100% coverage here, so some tests may seem unnecessary, but they cover some branch that could have been taken in the code. For instance, when I comment out these tests, these paths in the code coverage are untested.

![Screenshot from 2023-02-01 13-32-26](https://user-images.githubusercontent.com/3296972/216158091-e7afad77-f87f-4960-b2ac-d5ad238e8f40.png)

I think maybe there is a typo in your comment ("...at the very [first?] tests in this file"), but I didn't see other code covering the fromOptions() method.